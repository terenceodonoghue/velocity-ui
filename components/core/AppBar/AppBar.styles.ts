import { css, FunctionInterpolation, Theme } from '@emotion/react';

const appBar: FunctionInterpolation<Theme> = ({ palette, zIndex }) => css`
  display: flex;
  position: sticky;
  top: 0;
  border-bottom: solid 1px ${palette.neutral[300]};
  min-height: 80px;
  padding: 0 25px;
  background-color: ${palette.neutral[50]};
  z-index: ${zIndex.appBar};
`;

export default appBar;
