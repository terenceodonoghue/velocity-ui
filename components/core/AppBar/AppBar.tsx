import React, { FunctionComponent, HTMLAttributes } from 'react';
import appBar from './AppBar.styles';

const AppBar: FunctionComponent<HTMLAttributes<HTMLDivElement>> = ({
  children,
  ...props
}) => (
  <div css={appBar} data-testid="appbar" {...props}>
    {children}
  </div>
);

export default AppBar;
