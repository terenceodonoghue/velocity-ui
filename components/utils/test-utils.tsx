import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
