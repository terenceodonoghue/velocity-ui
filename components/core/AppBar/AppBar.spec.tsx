import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import AppBar from './AppBar';

describe('AppBar', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<AppBar>{content}</AppBar>);

    expect(getByTestId('appbar')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
