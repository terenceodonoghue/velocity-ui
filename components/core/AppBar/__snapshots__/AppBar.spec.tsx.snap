// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders content 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-position: sticky;
  position: sticky;
  top: 0;
  border-bottom: solid 1px #e4e8f0;
  min-height: 80px;
  padding: 0 25px;
  background-color: #fff;
  z-index: 1100;
}

<div>
  <div
    class="emotion-0"
    data-testid="appbar"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </div>
</div>
`;
